src/main/java/com/ust/sourcecourse/configuration/service/ProjectGroupService.java

```diff
@@ -176,8 +179,12 @@ public List<String> getTagsByGroup(Long uid) {
 
 	public List<ProjectGroupResponse> searchGroupsByTag(String tag) {
 		List<ProjectGroup> groups = projectGroupRepository.findAll();
-		return groups.stream().filter(group -> group.getTags() != null && group.getTags().contains(tag))
+		List<ProjectGroupResponse> matchingGroups= groups.stream().filter(group -> group.getTags() != null && group.getTags().contains(tag))
 				.map(group -> getProjectGroupresponse(group)).collect(Collectors.toList());
+		if (matchingGroups.isEmpty()) {
+	        throw new ResponseStatusException(HttpStatus.NOT_FOUND, "No project groups found with tag: " + tag);
```

Comment:
No need to return NOT_FOUND. Return 201-NO_CONTENT

Reply:
resolved